NeoML/Python/neoml/Blob.py

```diff
@@ -20,102 +20,137 @@
+    - Depth is the width of a 3-dimensional image
+    - Channels corresponds to channels for multi-channel image formats 
+        and is also used to work with one-dimensional vectors
+    
+    Data types
+    ---------
+    int and float data types are supported. Both are 32-bit.
+    """
 
     def __init__(self, internal):
-        """
+        """Creates a new blob.
         """
         if not type(internal) is PythonWrapper.Blob:
             raise ValueError('The `blob` must be PythonWrapper.Blob.')
         self._internal = internal
 
     @property
     def math_engine(self):
-        """
+        """The math engine which will perform the calculations.
         """
         return self._internal.math_engine()
 
     @property
     def shape(self):
-        """
+        """The blob shape: the array of 7 int elements with dimension lengths.
         """
         return self._internal.shape()
 
     @property
     def batch_len(self):
-        """
+        """Returns the BatchLength dimension.
         """
         return self._internal.batch_len()
 
     @property
     def batch_width(self):
-        """
+        """Returns the BatchWidth dimension.
         """
         return self._internal.batch_width()
 
     @property
     def list_size(self):
-        """
+        """Returns the ListSize dimension.
         """
         return self._internal.list_size()
 
     @property
     def height(self):
-        """
+        """Returns the Height dimension.
         """
         return self._internal.height()
 
     @property
     def width(self):
-        """
+        """Returns the Width dimension.
         """
         return self._internal.width()
 
     @property
     def depth(self):
-        """
+        """Returns the Depth dimension.
         """
         return self._internal.depth()
 
     @property
     def channels(self):
-        """
+        """Returns the Channels dimension.
         """
         return self._internal.channels()
 
     @property
     def size(self):
-        """
+        """Returns the blob total size: the product of all dimensions.
         """
         return self._internal.size()
 
     @property
     def object_count(self):
-        """
+        """Returns the number of objects in the blob, 
+        equal to BatchLength * BatchWidth * ListSize.
         """
         return self._internal.object_count()
 
     @property
     def object_size(self):
-        """
+        """Returns the size of one object, equal to
+        Height * Width * Depth * Channels.
         """
         return self._internal.object_size()
 
     def asarray(self, copy=False):
-        """
+        """Returns the contents of the blob as a multi-dimensional array, 
+        keeping only those dimensions that are more than 1 element long.
+        
+        Parameters
+        ---------
+        copy : bool, default=False
+            If True, the data will be copied. If False, the array will share
```

Comment:
If `copy=False` then the array _may_ share

(Reason: if blob is located in the GPU memory the copy will be created in any case because (unlike blob) array cannot be located in GPU memory)